typespec-tests/src/test/java/com/cadl/patch/PatchClientTest.java

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.cadl.patch;
+
+import com.azure.core.util.BinaryData;
+import com.cadl.patch.implementation.JsonMergePatchHelper;
+import com.cadl.patch.models.Fish;
+import com.cadl.patch.models.InnerModel;
+import com.cadl.patch.models.Resource;
+import com.cadl.patch.models.Shark;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+
+public class PatchClientTest {
+
+    @Test
+    public void testSerializationForNumbers() {
+        // patch not enabled
+        Resource resource = new Resource(new HashMap<>());
+        resource.setIntValue(null);
+        resource.setLongValue(null);
+        String json = BinaryData.fromObject(resource).toString();
+        Assertions.assertEquals("{\"map\":{}}", json);
```

Comment:
nit, don't do this string equal on testing JSON. Use e.g. `JsonNode node = mapper.readTree(json);` and validate the JsonNode.